Sources/HummingbirdAuth/Sessions/SessionAuthenticator.swift

```diff
@@ -18,7 +18,11 @@ import Hummingbird
 ///
 /// The `SessionAuthenticator` needs to have the ``SessionMiddleware`` before it in the middleware
 /// chain to extract session information for the request
-public struct SessionAuthenticator<Context: AuthRequestContext & SessionRequestContext, Repository: UserSessionRepository>: AuthenticatorMiddleware where Context.Session == Repository.Identifier {
+public struct SessionAuthenticator<
+    Identity: Sendable,
+    Context: AuthRequestContext<Identity> & SessionRequestContext,
```

Comment:
Again can the identity be extracted from the context instead of adding yet another generic parameter.